tests/tsl/tsl3/test.cs

```diff
@@ -0,0 +1,266 @@
+                var property = (PropertyInfo) type.GetMember("result").Single();
+                Assert.True(property.PropertyType == typeof(StringAccessor));
+            }
+        }
+
+    }
+
+    [Collection("TestServer Collection")]
+    public unsafe class ProtocolTest
+    {
+        private TrinityServerFixture Fixture { get; }
+        public ProtocolTest(TrinityServerFixture fixture) { Fixture = fixture; }
+
+        [Theory]
+        [InlineData(new byte[] {1, 2, 3, 4})]
+        [InlineData(new byte[] {2, 0, 4, 8})]
+        [InlineData(new byte[] {123, 124, 75, 43})]
+        [InlineData(new byte[] {77, 88, 9, 8})]
+        [InlineData(new byte[] {77, 88, 9, 8, 77, 88, 9, 8, 77, 88, 9, 8, 77, 88, 9, 8, })]
+        public void SynWithRsp_Test(byte[] nums)
+        {
+            Assert.True(nums.Length >= 4);
+            var numList = nums.ToList();
+            using (var writer = new ReqWriter())
+            {
+                writer.FieldBeforeList = numList[0];
+                writer.Nums.Add(numList[1]);
+                Console.WriteLine(string.Join(" ", numList.Skip(2).Select(Convert.ToInt32).ToList()));
+                writer.Nums.AddRange(numList.Skip(2).Select(Convert.ToInt32).ToList());
+                writer.Nums.RemoveAt(writer.Nums.Count - 1);
+                writer.FieldAfterList = numList.Last();
+                using (var response = Global.CloudStorage.TestSynWithRspToTestServer(Global.MyServerID, writer))
+                {
+                    Assert.Equal(string.Join(" ", numList), response.result);
+                }
+            }
+        }
+
+        [Fact]
+        public void Syn_Test()
+        {
+            using (var writer = new ReqWriter())
+            {
+                writer.FieldBeforeList = 2;
+                writer.Nums.Add(0);
+                writer.Nums.Add(7);
+                writer.FieldAfterList = 3;
+                Assert.Throws<IOException>(() => Global.CloudStorage.TestSynToTestServer(Global.MyServerID, writer));
+
+                using (var reader = new ReqReader(Utils.MakeCopyOfDataInReqWriter(writer), 0))
+                    Assert.Throws<NotImplementedException>(() => Fixture.Server.TestSynHandler(reader));
+
+                writer.FieldBeforeList = writer.FieldAfterList = 42;
+                using (var reader = new ReqReader(Utils.MakeCopyOfDataInReqWriter(writer), 0))
+                    Assert.Throws<ArgumentException>(() => Fixture.Server.TestSynHandler(reader));
+
+                writer.FieldBeforeList = 41;
+                writer.FieldAfterList = 42;
+                writer.Nums.Clear();
+                writer.Nums.AddRange(Enumerable.Repeat(1, 10).ToList());
+                using (var reader = new ReqReader(Utils.MakeCopyOfDataInReqWriter(writer), 0))
+                    Assert.Throws<InvalidDataException>(() => Fixture.Server.TestSynHandler(reader));
+            }
+        }
+
+
+        [Fact]
+        public void Asyn_Test()
```

Comment:
it could be very interesting to see if the server could handle a lot of async requests at the same time (reliably)